components/common/ShareDropdown.vue

```diff
@@ -0,0 +1,78 @@
+<template>
+  <NeoDropdown position="bottom-left" :mobile-modal="false">
+    <template #trigger="{ active }">
+      <NeoButton
+        :class="isMobileDevice ? 'icon-action' : ''"
+        :label="label"
+        :icon="icon"
+        :no-shadow="noShadow"
+        :active="active" />
+    </template>
+
+    <NeoDropdownItem
+      v-clipboard:copy="realworldFullPathShare"
+      @click="toast(String($t('toast.urlCopy')))">
+      {{ $t('share.copyLink') }}
+    </NeoDropdownItem>
+    <NeoDropdownItem @click="isModalActive = true">
+      {{ $t('share.qrCode') }}
+    </NeoDropdownItem>
+    <NeoDropdownItem
+      data-testid="gallery-item-share-dropdown-twitter"
+      @click="actionTwitterShare">
+      {{ $t('share.twitter') }}
+    </NeoDropdownItem>
+  </NeoDropdown>
+
+  <NeoModal :value="isModalActive" @close="isModalActive = false">
+    <div class="card">
+      <header class="card-header">
+        <p class="card-header-title">
+          {{ $t('share.qrCode') }}
+        </p>
+      </header>
+      <div class="card-content">
+        <QRCode
+          :text="realworldFullPathShare"
+          data-testid="gallery-item-share-dropdown-qrcode" />
+      </div>
+    </div>
+  </NeoModal>
+</template>
+
+<script setup lang="ts">
+import {
+  NeoButton,
+  NeoDropdown,
+  NeoDropdownItem,
+  NeoModal,
+} from '@kodadot1/brick'
+import { isMobileDevice } from '@/utils/extension'
```

Comment:
isMobileDevice checks if it's touch screen (or something similar cant quite remember)

is that truly the intention, or it should be about screen size?